File: Java/src/main/java/QuestionBank/Solution001_100/Solution26.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution26 {
    public int removeDuplicates(int[] nums) {
        int left = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[left]) {
                left++;
                nums[left] = nums[i];
            }
        }
        return left + 1;
    }


    public static void main(String[] args) {
        Solution26 solution26 = new Solution26();
        System.out.println(solution26.removeDuplicates(new int[]{1, 1, 2}));
    }
}

File: C++/src/Solution1_100/Solution26.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution26.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution26::removeDuplicates(vector<int> &nums) {
    if (nums.empty()) {
        return 0;
    }
    int i = 0;
    for (int j = 1; j < nums.size(); j++) {
        if (nums[j] != nums[i]) {
            i++;
            nums[i] = nums[j];
        }
    }
    return i + 1;
}

